src/dbnode/storage/series/series.go

```diff
@@ -288,26 +288,27 @@ func (s *dbSeries) Write(
 	unit xtime.Unit,
 	annotation []byte,
 	wOpts WriteOptions,
-) (bool, error) {
+) (bool, WriteType, error) {
+	var writeType WriteType
 	s.Lock()
 	matchUniqueIndex := wOpts.MatchUniqueIndex
 	if matchUniqueIndex {
 		if s.uniqueIndex == 0 {
-			return false, errSeriesMatchUniqueIndexInvalid
+			return false, writeType, errSeriesMatchUniqueIndexInvalid
 		}
 		if s.uniqueIndex != wOpts.MatchUniqueIndexValue {
 			// NB(r): Match unique index allows for a caller to
 			// reliably take a reference to a series and call Write(...)
 			// later while keeping a direct reference to the series
 			// while the shard and namespace continues to own and manage
 			// the lifecycle of the series.
-			return false, errSeriesMatchUniqueIndexFailed
+			return false, writeType, errSeriesMatchUniqueIndexFailed
```

Comment:
.. wow, we don't unlock here on these two early return paths.

Can you change this method to just use:
```go
s.Lock()
defer s.Unlock()
```

In go 1.14 this will become basically free.

Reply:
Wait, did we ever? Nice catch!